### PR TITLE
Error and warning correction

### PR DESCRIPTION
**Pull request by denis-maslovskiy**
Soon...

### CODE REVIEW

File: client/src/pages/Admin/AdminAddGame.tsx [hunk 1]
```diff
@@ -235,6 +272,34 @@ const AdminAddGame: React.FC = () => {
...
+                      className="form__input"
+                      label={input.label}
+                      variant="filled"
+                      onChange={handleChange}
+                      onBlur={handleBlur}
+                      name={input.name}
+                      type="number"
+                      value={values.discount || 0}
+                      InputProps={{ inputProps: { min: 0 } }}
+                      error={Boolean(values.discount < 0 || (!values.discount && values.discount !== 0))}
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminAddGame.tsx:296]: move to variable

File: client/src/pages/Admin/AdminAddGame.tsx [hunk 2]
```diff
@@ -235,6 +272,34 @@ const AdminAddGame: React.FC = () => {
...
+                      helperText={
+                        touched[input.name] &&
+                        Boolean(values.discount < 0 || (!values.discount && values.discount !== 0))
+                          ? "Discount must be greater than or equal to 0"
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminAddGame.tsx:300]: move to var

File: client/src/pages/Admin/AdminAddGame.tsx [hunk 3]
```diff
@@ -273,7 +344,9 @@ const AdminAddGame: React.FC = () => {
               type="submit"
               id="form-submit"
               className="add-game-button"
-              disabled={checksForButton(isSubmitting, errors, touched)}
+              disabled={
+                checksForButton(isSubmitting, errors, touched) || Boolean(!values.isDigital && !values.isPhysical)
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminAddGame.tsx:348]: move to var

File: client/src/pages/Admin/AdminEditGame.tsx [hunk 4]
```diff
@@ -358,37 +362,116 @@ const RenderGameForm = ({ initialGameData, deleteGameClickHandler, allGameAuthor
...
-                      />
+                      <FormControl error={!values.isDigital && !values.isPhysical}>
+                        <FormControlLabel
+                          // @ts-ignore
+                          control={<Checkbox {...field} color="primary" checked={values[`${input.name}`]} />}
+                          label={input.label}
+                          disabled={!Boolean(initialGameData.gameName)}
+                          className={input.name === "isDigital" ? "edit-game-digital-checkbox" : ""}
+                        />
+                        {!values.isDigital && !values.isPhysical && input.name === "isPhysical" && Boolean(initialGameData.gameName) && (
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminEditGame.tsx:381]: all long checks move to vars. It dificult to read in future and for other developers

File: client/src/pages/Admin/AdminEditGame.tsx [hunk 5]
```diff
@@ -358,37 +362,116 @@ const RenderGameForm = ({ initialGameData, deleteGameClickHandler, allGameAuthor
...
+                          variant="filled"
+                          type="number"
+                          InputProps={{ inputProps: { min: 0 } }}
+                          error={
+                            //@ts-ignore
+                            touched[input.name] &&
+                            Boolean(
+                              values.numberOfPhysicalCopies < 0 ||
+                                (!values.numberOfPhysicalCopies && values.numberOfPhysicalCopies !== 0)
+                            )
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminEditGame.tsx:410]: the same as above

File: client/src/pages/Admin/AdminEditGame.tsx [hunk 6]
```diff
@@ -358,37 +362,116 @@ const RenderGameForm = ({ initialGameData, deleteGameClickHandler, allGameAuthor
...
+                          }
+                          helperText={
+                            //@ts-ignore
+                            touched[input.name] &&
+                            values.isPhysical &&
+                            Boolean(
+                              values.numberOfPhysicalCopies < 0 ||
+                                (!values.numberOfPhysicalCopies && values.numberOfPhysicalCopies !== 0)
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminEditGame.tsx:418]: the same

File: client/src/pages/Admin/AdminEditGame.tsx [hunk 7]
```diff
@@ -358,37 +362,116 @@ const RenderGameForm = ({ initialGameData, deleteGameClickHandler, allGameAuthor
...
+                      <div className="form__div">
+                        <TextField
+                          {...field}
+                          label={input.label}
                           className="form__input"
                           variant="filled"
+                          disabled={!Boolean(initialGameData.gameName)}
+                          type="number"
+                          InputProps={{ inputProps: { min: 0 } }}
+                          error={Boolean(values.discount < 0 || (!values.discount && values.discount !== 0))}
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminEditGame.tsx:467]: you have the same check - > move to var

File: client/src/pages/Admin/AdminEditGame.tsx [hunk 8]
```diff
@@ -358,37 +362,116 @@ const RenderGameForm = ({ initialGameData, deleteGameClickHandler, allGameAuthor
...
+                          helperText={
+                            //@ts-ignore
+                            touched[input.name] &&
+                            Boolean(values.discount < 0 || (!values.discount && values.discount !== 0))
```
Comment (MeleshkoYuliya) [inline client/src/pages/Admin/AdminEditGame.tsx:471]: the same

**Comment by denis-maslovskiy** [comment]
Merge approved

**Pull request merged**